### PR TITLE
Pull dp config connect dp

### PR DESCRIPTION
**Pull request by geauxdojang**
This is a function to pull dp params to connect to dp.

### CODE REVIEW

File: R/dp_make_params.R [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+#' the private repository
+#' @return Named list of dp params
+#'
+#' @examples
+#' \dontrun{
+#' url <- "https://github.com/<USERNAME>/<REPOSITORY NAME>"
+#' dp_params <- dp_make_params(url = url, repo_token = Sys.getenv("GITHUB_PAT"))
+#' @export
+
+dp_make_params <- function(url, repo_token){
```
Comment (leslem) [inline R/dp_make_params.R:15]: What do others think of having this function look for an existing GITHUB_PAT variable as the default? Then you would only have to provide the repo_token arg if you want to override GITHUB_PAT, or don't have it set at all.

File: R/dp_make_params.R
Comment (leslem) [inline R/dp_make_params.R:1]: Looks good to me. A few comments to explain the general strategy (e.g. "Find config file on proper branch", or "Read config file from repo") would be helpful, but the excellent meaningful variable names do a great job making this code readable.

**Review by leslem — APPROVED** [review]

File: R/dp_make_params.R [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+#' @description execute and returns the value of function call given its textual
+#'  (dried) representation
+#' @param fn_called a function called
+#' @return value of the called function given its textual representation
+#' @examples \dontrun{
+#' fn_hydrate(fn_dry(sum(log(1:10))))
+#' }
+#' @keywords internal
+
+fn_hydrate <- function(dried_fn) {
```
Comment (camorosi) [inline R/dp_make_params.R:123]: Can we import this from `dpbuild` instead of redefining here?

File: R/dp_make_params.R [hunk 3]
```diff
@@ -103,7 +103,7 @@ dp_make_params <- function(url, repo_token){
 
   check_if_creds_empty <- unname(unlist((params_list$creds))) == ""
 
-  if (check_if_creds_empty) {
+  if (all(check_if_creds_empty)) {
     warning("Creds not found. Make sure that you set creds in your R environment as described daapr vignettes.")
```
Comment (leslem) [inline R/dp_make_params.R:107]:
```suggestion
    warning("Creds not found. Make sure that you set creds in your R environment as described in daapr vignettes.")
```

File: R/dp_make_params.R [hunk 4]
```diff
@@ -47,6 +40,13 @@ dp_make_params <- function(url, repo_token){
     retrieve_branch_name <- httr::GET(path_api_url_branches)
   }
 
+  http_status_code <- httr::status_code(retrieve_branch_name)
+
+  if (http_status_code != 200) {
+    stop(cli::cli_alert_danger("PAT is not found in R environment or not correctly set up.
```
Comment (leslem) [inline R/dp_make_params.R:46]:
```suggestion
    stop(cli::cli_alert_danger("GitHub PAT is not found in R environment or not correctly set up.
```

File: R/dp_make_params.R [hunk 5]
```diff
@@ -47,6 +40,13 @@ dp_make_params <- function(url, repo_token){
...
+        Make sure to pass or set up a token if this is a private repo. For e.g., Sys.setenv()"))
```
Comment (leslem) [inline R/dp_make_params.R:47]:
```suggestion
        Make sure to pass repo_token or set up GITHUB_PAT env var if this is a private repo. e.g. using Sys.setenv()"))
```

**Review by leslem — COMMENTED** [review]
Looks great! I just made a couple of suggestions for clearer error messages.

**Review by camorosi — APPROVED** [review]
This is working for me now on both github and enterprise. Looks good!

**Comment by amashadihossein** [comment]
> I've addressed/incorporated @leslem's feedback. Thanks. @amashadihossein @camorosi

I have 2 easy to address requests as I commented. Otherwise, it is looking great!

1. Eliminate the need for `stringr` import by replacing calls like 

      `stringr::str_split(string = "abc/def",pattern = "/")` 
      with base R calls like
      `strsplit(x = "abc/def",split = "/")`
      
      and then remove the stringr as a requirement. This is just to try to minimize dependencies and having to deal with additional maintenance. 

2. For consistency with `dp_init` change parameter name `url` to `github_repo_url`

**Comment by amashadihossein** [comment]
> > > I've addressed/incorporated @leslem's feedback. Thanks. @amashadihossein @camorosi
> > 
> > 
> > I have 2 easy to address requests as I commented. Otherwise, it is looking great!
> > 
> > 1. Eliminate the need for `stringr` import by replacing calls like
> >    `stringr::str_split(string = "abc/def",pattern = "/")`
> >    with base R calls like
> >    `strsplit(x = "abc/def",split = "/")`
> >    and then remove the stringr as a requirement. This is just to try to minimize dependencies and having to deal with additional maintenance.
> > 2. For consistency with `dp_init` change parameter name `url` to `github_repo_url`
> 
> @amashadihossein I addressed the [requested changes](https://github.com/amashadihossein/dpi/pull/18/commits/f8a650f5f1a188f4b44fcefe3a022a3701ef1d04). Can you review them? Thanks. @camorosi @leslem

`stringr` is still in the DESCRIPTION file. I am not seeing it used it anywhere though. Can it be taken out?

**Pull request merged**